### DISCUSSION TITLE
Secret use in specifying parts of settings.xml

### DISCUSSION THREAD

**Discussion by ksully-execute** [whelk-io/maven-settings-xml-action#158] [Q&A]
Can secrets (github organizational) be used to provide the values for the various pieces (servers, repositories, etc.)?
And could the secret have the entire content for the item, for example, for this entry in your examples:
repositories: '[{ "id": "some-repository", "url": "http://some.repository.url" }]'
could it done with:
repositories: ${{ secrets.MAVEN_REPOSITORIES}}
Thx.

**Comment by hiteshbal91dp** [comment]
@ksully-execute again I got some issue regarding using variables and secrets in actions

I am using following configuration where, I am getting repository and plugin repository config from variables and servers secret from github secrets and secrets are not being resolved it is placed as it is.

      - name: maven-settings-xml-action
        uses: whelk-io/maven-settings-xml-action@v20
        with:
          output_file: settings.xml
          servers: '[{"id":"central","username":"${secrets.JFROG_READ_USER}","password":"${secrets.JFROG_READ_PASS}"},{"id":"snapshots","username":"${secrets.JFROG_READ_USER}","password":"${secrets.JFROG_READ_PASS}"}]'
          repositories: ${{ vars.JFROG_REPOSITORIES_CONFIG }}
          plugin_repositories: ${{ vars.JFROG_PLUGIN_REPOSITORIES_CONFIG }}


Output

```
<servers>
    <server>
      <id>
        central
      </id>
      <username>
        ${secrets.JFROG_READ_USER}
      </username>
      <password>
        ${secrets.JFROG_READ_PASS}
      </password>
    </server>
    <server>
      <id>
        snapshots
      </id>
      <username>
        ${secrets.JFROG_READ_USER}
      </username>
      <password>
        ${secrets.JFROG_READ_PASS}
      </password>
    </server>
  </servers>
```

Please help me to resolve it.
